sound/soc/sof/core.c

```diff
@@ -197,6 +197,67 @@ static int sof_machine_check(struct snd_sof_dev *sdev)
 	return 0;
 }
 
+static int sof_select_ipc_and_paths(struct snd_sof_dev *sdev)
+{
+	struct snd_sof_pdata *plat_data = sdev->pdata;
+	struct sof_loadable_file_profile *base_profile = &plat_data->ipc_file_profile_base;
+	struct sof_loadable_file_profile out_profile;
+	struct device *dev = sdev->dev;
+	int ret;
+
+	/* check IPC support */
+	if (!(BIT(base_profile->ipc_type) & plat_data->desc->ipc_supported_mask)) {
+		dev_err(dev,
+			"ipc_type %d is not supported on this platform, mask is %#x\n",
+			base_profile->ipc_type, plat_data->desc->ipc_supported_mask);
+		return -EINVAL;
+	}
+
+	if (base_profile->ipc_type != plat_data->desc->ipc_default)
+		dev_info(dev,
+			 "Module parameter used, overriding default IPC %d to %d\n",
+			 plat_data->desc->ipc_default, base_profile->ipc_type);
+
+	if (base_profile->fw_path)
+		dev_dbg(dev, "Module parameter used, changed fw path to %s\n",
+			base_profile->fw_path);
+	else if (base_profile->fw_path_postfix)
+		dev_dbg(dev, "Path postfix appended to default fw path: %s\n",
+			base_profile->fw_path_postfix);
+
+	if (base_profile->fw_lib_path)
+		dev_dbg(dev, "Module parameter used, changed fw_lib path to %s\n",
+			base_profile->fw_lib_path);
+	else if (base_profile->fw_path_postfix)
+		dev_dbg(dev, "Path postfix appended to default fw_lib path: %s\n",
+			base_profile->fw_path_postfix);
```

Comment:
because they are firmware files, loaded to firmware and they must have matching postfixes in order to not mix invalid signature.
If the firmware is signed with, lt's say community key then the libraries should also have the same key used.

Reply:
This introduces no change in functionality.

Reply:
> because they are firmware files, loaded to firmware and they must have matching postfixes in order to not mix invalid signature. If the firmware is signed with, lt's say community key then the libraries should also have the same key used.

Yes but the community key is the exception to the rule. In generation the base firmware is signed with the Intel production key and the libs with the OEM key.

Reply:
> > because they are firmware files, loaded to firmware and they must have matching postfixes in order to not mix invalid signature. If the firmware is signed with, lt's say community key then the libraries should also have the same key used.
> 
> Yes but the community key is the exception to the rule. In generation the base firmware is signed with the Intel production key and the libs with the OEM key.

Oh, I see, hrm, how that should be handled then? We are certainly not taking this into account currently... OEM key is per OEM, right? So we should have Dell, Lenovo, HP, ASUS, etc postfixes for the library?

Reply:
probably yes, the postfix is likely to be modified on a per OEM basis - but IF OEMs decide to release such libraries with some sort of package.
It's also possible that the entire path for libraries is different, why would we put OEM-specific stuff until /lib/firmware/intel ?

Reply:
Right, the `fw_path_postfix` is only applied to the default path, if the path is provided via kernel parameter then it is used as it is. I think this makes it a sane rule (no different than what we have currently).

Reply:
we really need a fw_lib_path_postfix.